site/content/docs/howto/configure-auth-for-webapps.md

```diff
@@ -0,0 +1,347 @@
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+        - urn:ietf:params:oauth:grant-type:token-exchange
+      allowedScopes:
+        - openid
+        - offline_access
+        - pinniped:request-audience
+        - username
+        - groups
+    ```
+
+2. A web application which is allowed to use the Supervisor for authentication, but cannot perform actions on
+   Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+        - username
+        # "groups" can be excluded from this list when the webapp does
+        # not need to see the group memberships of the users.
+        - groups 
+    ```
+
+3. A web application which is allowed to use the Supervisor for authentication, but cannot see the username or
+   group memberships of the authenticated users, and cannot perform actions on Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+    ```
+
+## Create a client secret for the OIDCClient
+
+For each OIDCClient created by the Supervisor administrator, the administrator will also need to generate a client
+secret for the client. The client secrets are random strings auto-generated by the Supervisor upon request.
+The plaintext secret will only be returned once upon creation.
+
+```sh
+cat <<EOF | kubectl create -f -
+apiVersion: clientsecret.supervisor.pinniped.dev/v1alpha1
+kind: OIDCClientSecretRequest
+metadata:
+  name: client.oauth.pinniped.dev-my-webapp-client # the name of the OIDCClient
+  namespace: supervisor # the namespace of the OIDCClient
+spec:
+  generateNewSecret: true
+EOF
+```
+
+The server will respond with the newly generated client secret, e.g.:
+
+```
+NAMESPACE               NAME                                        SECRET     TOTAL
+pinniped-supervisor     client.oauth.pinniped.dev-my-webapp-client  abc123     1
+```
+
+Take care to make a note of the secret. After it has been returned by the create API, there is no other way to
+retrieve it in the future. The secret is not stored in plaintext on the server, which only stores a
+bcrypt-hashed version of the secret.
+
+This is the client secret that should be used, along with the client ID, by the web application when interacting
+with the Supervisor's OIDC token endpoint.
+
+## Rotating the client secret for an OIDCClient
+
+To facilitate rotating client secrets, an OIDCClient may have several active secrets. This enables the following process
+for the Supervisor administrator to change a client secret without causing web application downtime:
+
+1. Add a new, second secret to the OIDCClient by calling the create OIDCClientSecretRequest API again, as shown above.
```

Comment:
Should there be a "step 0" to remove the old client secrets? If there are several client secrets, then you may not be allowed to add a new one. This should probably also document that the max number of client secrets for any specific client is 5.

Reply:
The only awkward part about that is the user would have to "remove old client secrets" again at the end of the flow if they wanted to be sure to end with precisely 1 secret (though this isn't a huge deal since we prepend the new secret, so in theory, old secrets are dead secrets).